components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayJMSMessageListener.java

```diff
@@ -93,12 +98,13 @@ public void onMessage(Message message) {
             } else {
                 log.warn("Dropping the empty/null event received through jms receiver");
             }
-        } catch (JMSException e) {
+        } catch (JMSException | ArtifactSynchronizerException e) {
             log.error("JMSException occurred when processing the received message ", e);
         }
     }
 
-    private void handleNotificationMessage(String eventType, long timestamp, String encodedEvent) {
+    private void handleNotificationMessage(String eventType, long timestamp, String encodedEvent)
+            throws ArtifactSynchronizerException {
```

Comment:
Without throwing this exception from here, shall we add a try catch block and capture the exception inside this method

Reply:
https://github.com/wso2/carbon-apimgt/pull/9019/commits/6b56e3e82804d8002c3efac4f6c5e1de4286823d